tls/src/lib.rs

```diff
@@ -50,10 +48,6 @@ pub enum Error {
     #[error("io error: {0}")]
     Io(#[from] std::io::Error),
```

Comment:
As you surmised, John's doc is helpful here. There is no need to lose this context, it just gets a tad more tedious in usage. We should add specific errors for opening and reading files as described in https://github.com/oxidecomputer/omicron/blob/john/error-docs/docs/error-types-and-logging.adoc#2-defining-error-enums-with-thiserror.

We'd then change the `#[from]` to `#[source]` for the underlying error and add a `path` field that we could report in the `Display` impl with the same messages as we did previously in `with_context`.